### DISCUSSION TITLE
"Improve this page"/"Edit this page" links

### DISCUSSION THREAD

**Discussion by karenword** [carpentries/workbench#30] [Q&A] [answered]
During the Pre-beta and Beta phases, the Workbench or Styles versions of a lesson will exist as a "snapshot". As I understand it, new issues and PRs should be made only to the main repo throughout, and not to the 'snapshot' repo (except maybe for testing purposes). However, all lessons have a link to "Improve this page" (styles) or "Edit this page" (Workbench). If learners/trainees are presented with a "snapshot" version of a lesson, will clicking that button send them to the snapshot repo? If so, will there be any messaging (banners etc) to redirect them to the correct repo for making contributions?

Thanks!!

**Comment by zkamvar** [comment]
Thank you for opening this here! (note: For those that are unfamiliar with the overall strategy for the Beta Phase, You can find documentation at the [beta phase section](https://carpentries.github.io/workbench/beta-phase.html) of the workbench site.)

These are good questions and I have some answers, and realise that I may need to revise/update the strategy here.

> If learners/trainees are presented with a "snapshot" version of a lesson, will clicking that button send them to the snapshot repo?

I think a table would help here:

| phase    | live site               | snapshot                  | snapshot editable? | 
| -------- | ----------------------- | ------------------------- | ------------------ |
| pre-beta | `carpentries.github.io` | `preview.carpentries.org` | yes                |
|     beta | `preview.carpentries.org` | `carpentries.github.io` | no                 |

Short answer: yes for pre-beta stage, no for beta stage

Long answer:
 
### Pre-beta stage

In the pre-beta stage (two repositories, two sites), clicking on the "Edit this page" link on a workbench site (https://preview.carpentries.org/) will bring the reader to the page in the snapshot repository (in the https://github.com/fishtree-attempt organisation). At the moment, there is no indicator that changes in this repository will be discarded. 

Clicking on the "Improve this page" on the jekyll site (https://carpentries.github.io/) will bring the reader to the page in the live repository.

### Beta stage

In the beta stage (one repository, two sites), both sites will be deployed from the carpentries lesson program repository. 

The jekyll version will be snapshotted and it will be locked so that no modifications can be done. When François and I were planning this transition stage initially, [we envisioned](https://github.com/carpentries/styles/pull/637#issue-1191999364) that clicking on the "Improve this page" link on the would lead the reader to the home page of the repository (which would be the live site). 

The workbench version would lead to the page on the live site. 

> If so, will there be any messaging (banners etc) to redirect them to the correct repo for making contributions?

Maybe solution:

That being said, I believe can modify [the 'Edit this page' line in the HTML template](https://github.com/carpentries/varnish/blob/f966f6f2bfffef576ef22863e043d2f9f5841fe5/inst/pkgdown/templates/content-chapter.html#L27) to link to a page on the workbench website explaining the situation and gives a link to edit the page if they want. 

The only challenge for this is that outside of an episode, it's difficult to get the right page for the beta stage for a live edit.

**Comment by zkamvar** [comment, marked as answer]
An update to this:

>  If learners/trainees are presented with a "snapshot" version of a lesson, will clicking that button send them to the snapshot repo? If so, will there be any messaging (banners etc) to redirect them to the correct repo for making contributions?

Clicking on the button will send them to <https://carpentries.github.io/workbench/contributor/pre-beta.html> for the pre-beta stage and <https://carpentries.github.io/workbench/contributor/beta.html> for the beta stage. They will have the option to move on to make their edit from the link they click on. 

For example, R for Social Scientists is in pre-beta, so this link to [edit the introduction episode](https://carpentries.github.io/workbench/contributor/pre-beta.html?id=https://github.com/fishtree-attempt/r-socialsci/edit/main/episodes/00-intro.Rmd) from the snapshot will bring them to the explainer page with a link that forwards them on to edit the snapshot:

![screenshot that shows the landing page for the pre-beta stage. Text with a link at the bottom says "You may continue to propose changes to the temporary snapshot of this lesson"](https://user-images.githubusercontent.com/3639446/197649622-21de9687-5cd6-433c-ad6e-48b40bbb2cb3.png)


In 28 days, when R for Social Scientists enters the beta stage, the snapshot will be on the jekyll version of the site, and link to [edit the introduction episode](https://carpentries.github.io/workbench/contributor/beta.html?id=https://github.com/datacarpentry/r-socialsci/edit/main/_episodes_rmd/00-intro.Rmd) will again bring them to an explainer page that will forward them to edit the live page (not the snapshot):

![screenshot that shows the landing page for the beta stage. Text with a link at the bottom says "You may continue to propose changes to the live lesson"](https://user-images.githubusercontent.com/3639446/197649809-d73775a0-b749-44c3-b2c4-0cb2783dc90a.png)
